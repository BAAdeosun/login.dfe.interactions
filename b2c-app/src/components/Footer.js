import React from 'react'

export default function footer() {
    return (
        <div>
            I AM THE FOOTER
        </div>
    )
}
