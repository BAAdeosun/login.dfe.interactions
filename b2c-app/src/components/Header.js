import React from 'react'

function Header() {
    return (
        <header class="govuk-header " role="banner" data-module="govuk-header">
            <div class="govuk-header__container govuk-width-container">
                <div class="govuk-header__content">
                <a href="#" class="govuk-header__link govuk-header__link--service-name">
                    National Careers Service
                </a>
                <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
                <nav>
                    <ul id="navigation" class="govuk-header__navigation " aria-label="Top Level Navigation">
                    <li class="govuk-header__navigation-item govuk-header__navigation-item--active">
                        <a class="govuk-header__link" href="#1">
                        Job profiles
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#2">
                        Skills health check
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#3">
                        Find a course
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        Contact an adviser
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        About us
                        </a>
                    </li>
                    </ul>
                </nav>
                </div>
            </div>
        </header>
    )
}

export default Header;