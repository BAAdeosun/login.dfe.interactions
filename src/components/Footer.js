import Link from 'next/link'

const Footer = () => (
    <footer>
        This is the footer
    </footer>
)

export default Footer