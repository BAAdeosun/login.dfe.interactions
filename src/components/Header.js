import Link from 'next/link'

const Header = () => (
    <header>
        This is the header
    </header>
)

export default Header