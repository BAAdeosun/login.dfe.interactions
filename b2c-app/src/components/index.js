import Header from './Header/Header';
import Footer from './Footer/Footer';
import B2C from './B2C/B2C';
import HeaderTitle from './HeaderTitle/HeaderTitle';
import PageNavigation from './PageNavigation/PageNavigation';
import NavigationLink from './NavigationLink/NavigationLink';
import Breadcrumbs from './Breadcrumbs/Breadcrumbs';
import MainContainer from './MainContainer/MainContainer';
import PageTitle from './PageTitle/PageTitle';
import FooterNavigation from './FooterNavigation/FooterNavigation';
import FooterLink from './FooterLink/FooterLink';
import PasswordHelpContainer from './PasswordHelp/PasswordHelpContainer';
import PasswordHelp from './PasswordHelp/PasswordHelp';
import PageLevelErrorContainer from './PageLevelErrorContainer/PageLevelErrorContainer';
import CreateNewPassword from './CreateNewPassword/CreateNewPassword';
import DateOfBirth from './DateOfBirth/DateOfBirth';
import TermsAndConditions from './TermsAndConditions/TermsAndConditions';
import ButtonLink from './ButtonLink/ButtonLink';
import Link from './Link/Link';
import Paragraph from './Paragraph/Paragraph';
import Postcode from './Postcode/Postcode';
import InputField from './InputField/InputField';

export default {
    Header: Header,
    Footer: Footer,
    B2C: B2C,
    HeaderTitle: HeaderTitle,
    PageNavigation: PageNavigation,
    NavigationLink: NavigationLink,
    Breadcrumbs: Breadcrumbs,
    MainContainer: MainContainer,
    PageTitle: PageTitle,
    FooterNavigation: FooterNavigation,
    FooterLink: FooterLink,
    PasswordHelpContainer: PasswordHelpContainer,
    PasswordHelp: PasswordHelp,
    PageLevelErrorContainer: PageLevelErrorContainer,
    CreateNewPassword: CreateNewPassword,
    DateOfBirth: DateOfBirth,
    TermsAndConditions: TermsAndConditions,
    ButtonLink: ButtonLink,
    Link: Link,
    Paragraph: Paragraph,
    Postcode: Postcode,
    InputField: InputField
}