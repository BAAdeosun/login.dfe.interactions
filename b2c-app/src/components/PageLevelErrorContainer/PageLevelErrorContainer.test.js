import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import components from '..';

it('renders without crashing', () => {
    shallow(<components.PageLevelErrorContainer />);
});

it('renders correctly without props', () => {
    const tree = renderer
        .create(<components.PageLevelErrorContainer />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

describe('when there are errors', () => {

    const errorItems = [
        {
            text: 'test 1'
        },
        {
            id: 'id2',
            text: 'test 2'
        }
    ];

    it('renders correctly', () => {

        const tree = renderer
            .create(<components.PageLevelErrorContainer errorItems={errorItems} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with the optional error summary', () => {

        errorItems.push(
            {
                text: 'test 3',
                showSummaryText: true
            }
        )

        const summaryTextContent = <p>error summary</p>

        const tree = renderer
            .create(<components.PageLevelErrorContainer errorItems={errorItems} summaryTextContent={summaryTextContent} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('when there are B2C errors', () => {

    const b2cErrors = ['error 1', 'error 2'];

    it('renders correctly', () => {
        const tree = renderer
            .create(<components.PageLevelErrorContainer showB2CErrors={true} />);
        expect(tree.toJSON()).toMatchSnapshot();

        //add B2C errors
        tree.root.instance.setState({ b2cErrors: b2cErrors });

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('renders correctly with showB2CErrors set to false', () => {
        const tree = renderer
            .create(<components.PageLevelErrorContainer showB2CErrors={false} />);
        expect(tree.toJSON()).toMatchSnapshot();

        //add B2C errors
        tree.root.instance.setState({ b2cErrors: b2cErrors });

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('sets the errors added by B2C to the state in the mutation callback', () => {

        //prepare mutation data to simulate adding/removing B2C errors
        const errorElement = document.createElement('div');
        errorElement.classList.add('error');
        errorElement.style.display = 'none';
        document.body.appendChild(errorElement);

        //set mutation list to simulate B2C showing or hiding errors
        const b2cErrorMutations = [
            {
                target: errorElement
            }
        ];
        let b2cErrorMutationList = b2cErrorMutations[Symbol.iterator]();

        //Mount spinner
        const wrapper = mount(<components.PageLevelErrorContainer />);

        //b2c errors is an empty array initially
        expect(wrapper.state().b2cErrors.length).toEqual(0);

        //call mutation callback, error does not have pageLevel class so there are no errors to add
        wrapper.instance().pageLevelErrorCallback(b2cErrorMutationList);
        //check b2cErrors is still empty
        expect(wrapper.state().b2cErrors.length).toEqual(0);

        //add pageLevel class to error, still display: 'none'
        b2cErrorMutationList = b2cErrorMutations[Symbol.iterator]();
        errorElement.classList.add('pageLevel');
        //call mutation callback, error has display none so still no errors to add
        wrapper.instance().pageLevelErrorCallback(b2cErrorMutationList);
        //check b2cErrors is still empty
        expect(wrapper.state().b2cErrors.length).toEqual(0);

        //set error as visible
        errorElement.style.display = 'block';
        b2cErrorMutationList = b2cErrorMutations[Symbol.iterator]();
        //call mutation callback
        wrapper.instance().pageLevelErrorCallback(b2cErrorMutationList);
        //check b2cErrors has one item now
        expect(wrapper.state().b2cErrors.length).toEqual(1);

        //set display: 'none' again, error should be removed
        errorElement.style.display = 'none';
        b2cErrorMutationList = b2cErrorMutations[Symbol.iterator]();
        //call mutation callback
        wrapper.instance().pageLevelErrorCallback(b2cErrorMutationList);
        //check b2cErrors has no items
        expect(wrapper.state().b2cErrors.length).toEqual(0);
    });
});