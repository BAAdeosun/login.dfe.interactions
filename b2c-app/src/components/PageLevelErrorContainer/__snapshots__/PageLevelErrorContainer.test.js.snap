// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly without props 1`] = `
<div
  className="pageLevelErrorContainer hide"
>
  <div
    aria-labelledby="error-summary-title"
    className="govuk-error-summary"
    data-module="govuk-error-summary"
    role="alert"
    tabIndex="-1"
  >
    <h2
      className="govuk-error-summary__title"
      id="error-summary-title"
    >
      There is a problem
    </h2>
    <div
      className="govuk-error-summary__body"
    >
      <ul
        className="govuk-list govuk-error-summary__list"
        id="errorSummaryItems"
      />
    </div>
  </div>
</div>
`;

exports[`when there are B2C errors renders correctly 1`] = `
<div
  className="pageLevelErrorContainer hide"
>
  <div
    aria-labelledby="error-summary-title"
    className="govuk-error-summary"
    data-module="govuk-error-summary"
    role="alert"
    tabIndex="-1"
  >
    <h2
      className="govuk-error-summary__title"
      id="error-summary-title"
    >
      There is a problem
    </h2>
    <div
      className="govuk-error-summary__body"
    >
      <ul
        className="govuk-list govuk-error-summary__list"
        id="errorSummaryItems"
      />
    </div>
  </div>
</div>
`;

exports[`when there are B2C errors renders correctly 2`] = `
<div
  className="pageLevelErrorContainer show"
>
  <div
    aria-labelledby="error-summary-title"
    className="govuk-error-summary"
    data-module="govuk-error-summary"
    role="alert"
    tabIndex="-1"
  >
    <h2
      className="govuk-error-summary__title"
      id="error-summary-title"
    >
      There is a problem
    </h2>
    <div
      className="govuk-error-summary__body"
    >
      <ul
        className="govuk-list govuk-error-summary__list"
        id="errorSummaryItems"
      >
        <li>
          error 1
        </li>
        <li>
          error 2
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`when there are B2C errors renders correctly with showB2CErrors set to false 1`] = `
<div
  className="pageLevelErrorContainer hide"
>
  <div
    aria-labelledby="error-summary-title"
    className="govuk-error-summary"
    data-module="govuk-error-summary"
    role="alert"
    tabIndex="-1"
  >
    <h2
      className="govuk-error-summary__title"
      id="error-summary-title"
    >
      There is a problem
    </h2>
    <div
      className="govuk-error-summary__body"
    >
      <ul
        className="govuk-list govuk-error-summary__list"
        id="errorSummaryItems"
      />
    </div>
  </div>
</div>
`;

exports[`when there are B2C errors renders correctly with showB2CErrors set to false 2`] = `
<div
  className="pageLevelErrorContainer hide"
>
  <div
    aria-labelledby="error-summary-title"
    className="govuk-error-summary"
    data-module="govuk-error-summary"
    role="alert"
    tabIndex="-1"
  >
    <h2
      className="govuk-error-summary__title"
      id="error-summary-title"
    >
      There is a problem
    </h2>
    <div
      className="govuk-error-summary__body"
    >
      <ul
        className="govuk-list govuk-error-summary__list"
        id="errorSummaryItems"
      />
    </div>
  </div>
</div>
`;

exports[`when there are errors renders correctly 1`] = `
<div
  className="pageLevelErrorContainer show"
>
  <div
    aria-labelledby="error-summary-title"
    className="govuk-error-summary"
    data-module="govuk-error-summary"
    role="alert"
    tabIndex="-1"
  >
    <h2
      className="govuk-error-summary__title"
      id="error-summary-title"
    >
      There is a problem
    </h2>
    <div
      className="govuk-error-summary__body"
    >
      <ul
        className="govuk-list govuk-error-summary__list"
        id="errorSummaryItems"
      >
        <li>
          <p>
            test 1
          </p>
        </li>
        <li>
          <a
            href="#"
            onClick={[Function]}
          >
            test 2
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`when there are errors renders correctly with the optional error summary 1`] = `
<div
  className="pageLevelErrorContainer show"
>
  <div
    aria-labelledby="error-summary-title"
    className="govuk-error-summary"
    data-module="govuk-error-summary"
    role="alert"
    tabIndex="-1"
  >
    <h2
      className="govuk-error-summary__title"
      id="error-summary-title"
    >
      There is a problem
    </h2>
    <div
      id="errorSummaryText"
    >
      <p>
        error summary
      </p>
    </div>
    <div
      className="govuk-error-summary__body"
    >
      <ul
        className="govuk-list govuk-error-summary__list"
        id="errorSummaryItems"
      >
        <li>
          <p>
            test 1
          </p>
        </li>
        <li>
          <a
            href="#"
            onClick={[Function]}
          >
            test 2
          </a>
        </li>
        <li>
          <p>
            test 3
          </p>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
